src/commons/addresses.ts

```diff
@@ -105,3 +105,79 @@ export function getLiquidityTrackingTokenAddresses(): string[] {
   log.warning('no liquidity tracking token address for unsupported network {}', [network])
   return []
 }
+
+export function getUsdcNativeCurrencyWrapperPairAddress(): string {
+  let network = dataSource.network() as string
+  // not using a switch-case because using strings is not yet supported (only u32)
+  if (network == 'mainnet') return '0x98f29f527c8e0ecc67a3c2d5567833bee01f2a12'
+  if (network == 'rinkeby') return '0xf3a261b601cd965cd18a5618c436a899c9abed51'
+  if (network == 'xdai') return ADDRESS_ZERO
+  if (network == 'arbitrum-one') return '0x403b1405d8caffc1cc5032cc82aa135d2481d0cf'
+  if (network == 'arbitrum-rinkeby') return ADDRESS_ZERO
+  log.warning('no usdc native currency wrapper pair address for unsupported network {}', [network])
+  return ADDRESS_ZERO
+}
+
+export function getDaiNativeCurrencyWrapperPairAddress(): string {
+  let network = dataSource.network() as string
+  // not using a switch-case because using strings is not yet supported (only u32)
+  if (network == 'mainnet') return '0x7515be43d16f871588adc135d58a9c30a71eb34f'
+  if (network == 'rinkeby') return ADDRESS_ZERO
+  if (network == 'xdai') return ADDRESS_ZERO
+  if (network == 'arbitrum-one') return ADDRESS_ZERO
+  if (network == 'arbitrum-rinkeby') return '0x1bb34bc1967e0cbe1c9008e5390c6d426357549d'
+  log.warning('no dai native currency wrapper pair address for unsupported network {}', [network])
+  return ADDRESS_ZERO
+}
+
+export function getUsdtNativeCurrencyWrapperPair(): string {
+  let network = dataSource.network() as string
+  // not using a switch-case because using strings is not yet supported (only u32)
+  if (network == 'mainnet') return '0x83dd8227c5ef121f2ae99c6f1df0aa9e914448ce'
+  if (network == 'rinkeby') return ADDRESS_ZERO
+  if (network == 'xdai') return ADDRESS_ZERO
+  if (network == 'arbitrum-one') return ADDRESS_ZERO
+  if (network == 'arbitrum-rinkeby') return ADDRESS_ZERO
+  log.warning('no usdt native currency wrapper pair address for unsupported network {}', [network])
+  return ADDRESS_ZERO
+}
+
+/**
+ * Swapr token address, supported network: Mainnet, Arbitrum One, xDAI, Rinkeby, and Arbitrum Rinkeby
+ */
+export function getSwaprTokenAddress(): string {
+  let network = dataSource.network() as string
+  // Production
+  if (network == 'mainnet') return '0x6cacdb97e3fc8136805a9e7c342d866ab77d0957'
+  if (network == 'arbitrum-one') return '0xde903e2712288a1da82942dddf2c20529565ac30'
+  if (network == 'xdai') return '0x532801ED6f82FFfD2DAB70A19fC2d7B2772C4f4b'
+  /**
+   * @todo get the correct addreses
```

Comment:
Why this todo?

Reply:
It was copied from the original subgraph repo. It's fixed now :)